src/sleep/sleep.c

```diff
@@ -263,7 +263,17 @@ static int execute_s2h(usec_t hibernate_delay_sec) {
                 return 0;
 
         /* If woken up after alarm time, hibernate */
-        return execute(hibernate_modes, hibernate_states);
+        r = execute(hibernate_modes, hibernate_states);
+        if (r < 0) {
+                log_notice("Couldn't hibernate, will try to suspend again.");
+                r = execute(suspend_modes, suspend_states);
+                if (r < 0) {
+                        log_notice("Could neither hibernate nor suspend again, giving up.");
+                        return r;
```

Comment:
I think we should initialize shutdown here. But let's discuss that in a separate PR.

Reply:
hmm, what? "initialize"? do you mean "initiate"? but that sounds quite drastic to me, no? I mean, if we go back to suspend we have the chance to save the user's data, if we shutdown right away that's very unlikely...

Reply:
Right, I meant "initiate". I'm worried that after s2h the machine might be in a bag or something like that and might overheat. But maybe we don't need to worry, since the kernel should kill the machine if it crosses some overheat threshold. So nevermind, we don't need to do anything.